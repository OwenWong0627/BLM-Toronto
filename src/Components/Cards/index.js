import React from 'react';
import './Cards.css';
import CardItem from '../CardItem';

/**
 * This component represents the card section of the home page where the card items will showcase previews of
 * the other two pages and provide interactive to redirect users to one of the two pages
 */
function Cards() {
   return (
      <div data-testid="cards" className='cards'>
         <h1>Check Out The App Features & Functionalities!</h1>
         <div className='cards__container'>
            <div className='cards__wrapper'>
               <ul data-testid="card-list" className='cards__items'>
                  <CardItem
                     src='/Map_Preview.png'
                     text='Explore The Map to Find Your Local Black-owned Business'
                     label='Map'
                     to='/find-business'
                  />
                  <CardItem
                     src='/List_Business_Preview.png'
                     text='Check Out This Page to Find Your Local Black-owned Virtual Business'
                     label='Pagination'
                     to='/list-business/'
                  />
               </ul>
            </div>
         </div>
      </div>
   );
}

export default Cards;