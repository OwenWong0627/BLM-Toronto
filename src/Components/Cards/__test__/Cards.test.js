import React from 'react';
import ReactDOM from 'react-dom';
import Cards from '..';

import { BrowserRouter as Router } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup);
it('Renders without crashing', () => {
   const div = document.createElement("div");
   ReactDOM.render(<Router><Cards /></Router>, div);
});

describe('Renders two CardItem components', () => {
   it('Checks if the cards component has the static title outlined in the component', () => {
      const { getByTestId } = render(<Router><Cards /></Router>);
      expect(getByTestId("cards")).toHaveTextContent("Check Out The App Features & Functionalities!");
   });
   it('Checks if the first card item with the Map description is rendered', () => {
      const { getByTestId } = render(<Router><Cards /></Router>);
      expect(getByTestId("cards")).toHaveTextContent("Explore The Map to Find Your Local Black-owned Business");
   });
   it('Checks if the second card item with the List Virtual Businesses description is rendered', () => {
      const { getByTestId } = render(<Router><Cards /></Router>);
      expect(getByTestId("cards")).toHaveTextContent("Check Out This Page to Find Your Local Black-owned Virtual Business");
   });
});

it('matches snapshot', () => {
   const tree = renderer.create(<Router><Cards /></Router>).toJSON();
   expect(tree).toMatchSnapshot();
});