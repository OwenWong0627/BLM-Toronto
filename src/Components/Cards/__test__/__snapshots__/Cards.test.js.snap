// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="cards"
  data-testid="cards"
>
  <h1>
    Check Out The App Features & Functionalities!
  </h1>
  <div
    className="cards__container"
  >
    <div
      className="cards__wrapper"
    >
      <ul
        className="cards__items"
      >
        <li
          className="cards__item"
        >
          <a
            className="cards__item__link"
            data-testid="card-item"
            href="/find-business"
            onClick={[Function]}
          >
            <figure
              className="cards__item__pic-wrap"
              data-category="Map"
              data-testid="card-item-label"
            >
              <img
                alt="Page Previews"
                className="cards__item__img"
                data-testid="card-item-img"
                src="/Map_Preview.png"
              />
            </figure>
            <div
              className="cards__item__info"
            >
              <h2
                className="cards__item__text"
              >
                Explore The Map to Find Your Local Black-owned Business
              </h2>
            </div>
          </a>
        </li>
        <li
          className="cards__item"
        >
          <a
            className="cards__item__link"
            data-testid="card-item"
            href="/list-business/"
            onClick={[Function]}
          >
            <figure
              className="cards__item__pic-wrap"
              data-category="Pagination"
              data-testid="card-item-label"
            >
              <img
                alt="Page Previews"
                className="cards__item__img"
                data-testid="card-item-img"
                src="/List_Business_Preview.png"
              />
            </figure>
            <div
              className="cards__item__info"
            >
              <h2
                className="cards__item__text"
              >
                Check Out This Page to Find Your Local Black-owned Virtual Business
              </h2>
            </div>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
